### PR TITLE
Minor refactoring of Object/abstract.c (UNARY_FUNC macro and more cases for BINARY_FUNC)

### PR DESCRIPTION
**Pull request by skirpichev**
* Use BINARY_FUNC macro for some remaining ops
* Add UNARY_FUNC macro to define unary PyNumber_* functions

------------

// from #111996

### CODE REVIEW

**Comment by skirpichev** [comment]
CC @serhiy-storchaka, this was abstract.c diff from the referenced pr.  I hope this can skip issue and news entry...

File: Objects/abstract.c [hunk 1]
```diff
@@ -1379,73 +1344,27 @@ _PyNumber_InPlacePowerNoMod(PyObject *lhs, PyObject *rhs)
...
+        }                                                                \
+                                                                         \
+        PyNumberMethods *m = Py_TYPE(o)->tp_as_number;                   \
+        if (m && m->op) {                                                \
+            PyObject *res = (*m->op)(o);                                 \
+            assert(_Py_CheckSlotResult(o, #meth_name, res != NULL));     \
+            return res;                                                  \
+        }                                                                \
+                                                                         \
+        return type_error("bad operand type for unary -: '%.200s'", o);  \
```
Comment (serhiy-storchaka) [inline Objects/abstract.c]: Error message is wrong.

**Review by serhiy-storchaka — COMMENTED** [review]
It deserves 4 or 5 separate issues.

File: Objects/abstract.c (hunk 1 above)
Comment (skirpichev) [reply Objects/abstract.c]: Oops, forgot last argument.

**Comment by skirpichev** [comment]
> It deserves 4 or 5 separate issues.

Huh.  Is it worth or code should be kept intact?

Reusing old or adding new macros is a secondary issue.  All others related to a dead code (for different reasons).  So, having something in common.

**Comment by skirpichev** [comment]
@serhiy-storchaka, now this has only UNARY/BINARY_FUNC stuff.  Probably this not worth an issue and can skip news.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**